assets/javascripts/discourse/initializers/discourse-reactions.gjs

```diff
@@ -128,21 +128,26 @@ function initializeDiscourseReactions(api) {
 
         get label() {
           const count = this.notification.data.count;
-          const username = this.username;
+          const nameOrUsername = this.siteSettings.prioritize_full_name_in_ux
+            ? this.notification.data.display_name ||
+              this.notification.data.username
+            : this.username;
 
           if (!count || count === 1 || !this.notification.data.username2) {
-            return username;
+            return nameOrUsername;
           }
-
           if (count > 2) {
             return i18n("notifications.reaction_multiple_users", {
-              username,
+              username: nameOrUsername,
               count: count - 1,
             });
           } else {
+            const nameOrUsername2 = this.siteSettings.prioritize_full_name_in_ux
+              ? this.notification.data.name2 || this.notification.data.username2
```

Comment:
```suggestion
              ? this.notification.data.name2 || formatUsername(this.notification.data.username2)
```

(or possibly extract `formatUsername(this.notification.data.username2)` into a variable and use it in the 2 spots in this ternary condition, up to you)